src/main/java/net/mcreator/element/parts/gui/Sprite.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.parts.gui;
+
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.ui.wysiwyg.WYSIWYGEditor;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.util.image.ImageUtils;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.references.TextureReference;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class Sprite extends GUIComponent {
+
+	@TextureReference(TextureType.SCREEN) public String sprite;
+
+	// Each sprite width and height
+	public int spriteWidth, spriteHeight;

```

Comment:
See Image GUI component for example. We don't store width and height but compute if from image

Reply:
I don't quite understand what you mean. For example, the selected texture is 128x16, and the size of each sprite is 16x16, which the user inputs.
The only thing that comes to my mind is that instead of getting the height and the width of the sprite from the user, we get the number of sprites.

Reply:
Aha ok, I see, I wrongly assumed sprites can only be square, but this is ofc not the case

Reply:
Currently, sprites must be in a single row or column. But if it needs to have more than one row or column, we would need two indices to select each sprite, one for x-axis and one for y-axis. However, I think this would make it a bit complicated for the user to select their sprite using two procedures

Reply:
Currently, sprites must be in a single row or column. But if it needs to have more than one row or column, we would need two indices to select each sprite, one for x-axis and one for y-axis. However, I think this would make it a bit complicated for the user to select their sprite using two procedures

Reply:
Yes, that would be too complicated. In this case, a number of sprites input may be the best idea